optimade/validator/validator.py

```diff
@@ -135,11 +137,22 @@ def __init__(  # pylint: disable=too-many-arguments
         if client:
             self.client = client
             self.base_url = self.client.base_url
+            # If a custom client has been provided, try to set custom headers if they have been specified,
+            # but do not overwrite any existing attributes
+            if http_headers:
+                if not hasattr(self.client, "headers"):
+                    self.client.headers = http_headers
+                else:
+                    print_warning(
+                        f"Not using specified request headers {http_headers} with custom client {self.client}."
```

Comment:
Is there no way to add the headers to the client? E.g., using the `dict.update()` method? Or is this a conceptual choice?
I would think that even though a custom client is passed here, if `http_headers` is also passed in, then it should be added no matter what, right?

Reply:
Currently the client is allowed to be `Any` class with the expectation that it implements a `.get()` method, but we could consider restricting that further. As the `optimade.validator.utils.Client` is the only officially supported class, and the only other client we use is the TestClient, I'm not sure how much we need to worry about this (beyond making sure we don't overwrite data).

Reply:
Okay, fair enough. Then this is the perfect solution I think. Great! :)